scripts/template_matcher.py

```diff
@@ -59,15 +84,58 @@ def _compute_prediction(self, k, img, kp, des):
         kp: keypoints from scene image,   des: descriptors from scene image
         """
 
-        # TODO: find corresponding points in the input image and the templae image
-        #       put keypoints from template image in template_pts
-        #       put corresponding keypoints from input image in img_pts
-
-
-        #TODO: change img to img_T once you do the homography transform
-        visual_diff = compare_images(img, self.signs[k])
+        # Get list of matches between keypoints based on destriptors
+        matches = self.matcher.knnMatch(self.descs[k],des,k=2)
+        # Convert into point form
+        matches = [(m.queryIdx, m.trainIdx) for m, n in matches]
+
+        # Convert those points to points findHomography can handle
+        template_pts = np.zeros((len(matches),2))
+        img_pts = np.zeros((len(matches),2))
+        for idx in range(len(matches)):
+            match = matches[idx]
+            template_pts[idx,:] = self.kps[k][match[0]].pt
+            img_pts[idx,:] = kp[match[1]].pt
+
+        # Transform input image so that it matches the template image as well as possible
+        M, mask = cv2.findHomography(img_pts, template_pts, cv2.RANSAC, self.good_thresh)
```

Comment:
So we were intending good_thresh to be used as a ratio threshold, and findHomography has a different ransac_threshold which was supposed to be used here.  

See the ransacReprojThreshold variable for more explanation on what this 4th parameter is: http://docs.opencv.org/2.4/modules/calib3d/doc/camera_calibration_and_3d_reconstruction.html?highlight=findhomography#findhomography